File: ruby/lib/resp_bench/command/impl/ping_command.rb
# frozen_string_literal: true

module RespBench
  module Command
    module Impl
      # PING command implementation.
      class PingCommand < Command
        def execute(client, _key_generator)
          result = client.ping
          CommandResult.new(
            command_name: @name,
            latency_micros: result.latency_micros,
            success: result.success?
          )
        end
      end
    end
  end
end

File: python/src/resp_bench/command/impl/ping_command.py
"""PING command (ignores the generated key)."""

from __future__ import annotations

from typing import Optional

from ...client.benchmark_client import AsyncBenchmarkClient
from ..command import Command, CommandResult


class PingCommand(Command):
    async def execute(self, client: AsyncBenchmarkClient, key: Optional[str]) -> CommandResult:
        result = await client.ping()
        return CommandResult(
            command_name=self.name,
            latency_micros=result.latency_micros,
            success=result.success,
        )
